templates/client-daemonset.yaml

```diff
@@ -6,24 +6,25 @@ metadata:
   name: {{ template "consul.fullname" . }}
   namespace: {{ .Release.Namespace }}
   labels:
-    app: {{ template "consul.name" . }}
-    chart: {{ template "consul.chart" . }}
-    heritage: {{ .Release.Service }}
-    release: {{ .Release.Name }}
+    helm.sh/chart: {{ include "consul.chart" . }}
+    app.kubernetes.io/name: {{ include "consul.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/version: {{ .Chart.Version | quote }}
```

Comment:
This prevents upgrades because `labels` is an immutable field for daemonset and statefulsets

Reply:
Ok, what do you recommend this should be changed to?

Reply:
Remove any label that will change if the chart version is upgraded.